### PR TITLE
Feature: Add option to filter profiles by archive type for `xcode-project use-profiles` 

### PR DESCRIPTION
**Pull request by priitlatt**
Export options properties generated by `xcode-project use-profiles` can have ambiguous archive method and signing certificate settings. This happens when provisioning profiles of different types are found during the lookup and profiles with different certificates are assigned to Xcode project targets.

For instance when both development and distribution profiles are present in the profiles directory (by default in`$HOME/Library/MobileDevice/Provisioning Profiles`) and all profiles have certificates in the keychain, then it is completely valid that some targets in Xcode project are assigned development profiles and others are assigned App Store / distribution profiles. As a result of that it is not possible to reliably decide which archive method, and consequently certificate, should be chosen for the generated export options.

To overcome this problem, add an option to limit provisioning profile lookup to match only those profiles that can be used for specified achive method (`ad-hoc`, `app-store`, `enterprise` or `development`). This is achieved by new optional argument `--archive-method` for action `xcode-project use-profiles`. In case archive method is not defined, the behaviour of the command will remain as it used to be.

**Updated actions**
- `xcode-project use-profiles`

**Usage examples**

<details><summary>Signing method and certificate choice is ambiguous if archive method is not specified.</summary> 
<p>
Example usage of 'xcode-project use-profiles' without '--archive-method'
</p>
</details>

<details><summary>Setting <code>--archive-method development</code> allows only development profiles and certificates.</summary> 
<p>
Example usage of 'xcode-project use-profiles' with '--archive-method development'
</p>
</details>

<details><summary>Setting <code>--archive-method ad-hoc</code> allows only ad-hoc profiles and distribution certificates.</summary> 
<p>
Example usage of 'xcode-project use-profiles' with '--archive-method ad-hoc'
</p>
</details>

<details><summary>Setting <code>--archive-method app-store</code> allows only distribution profiles and certificates.</summary> 
<p>
Example usage of 'xcode-project use-profiles' with '--archive-method app-store'
</p>
</details>

<details><summary>Using <code>--archive-method</code> value for which profiles are not found yields empty code signing settings.</summary> 
<p>
Example usage of 'xcode-project use-profiles' with '--archive-method enterprise'
</p>
</details>

<details><summary>Using invalid <code>--archive-method</code> value raises an error.</summary> 
<p>
Example usage of 'xcode-project use-profiles' with invalid '--archive-method' value
</p>
</details>

### CODE REVIEW

**Comment by priitlatt** [comment]
Changelog is still missing.

**Review by mikhail-tokarev — APPROVED** [review]
will be nice to add version description but otherwise LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Version 0.25.0
+-------------
+
+TBD
```
[the remark this answers is not in the document]
Comment (priitlatt) [reply CHANGELOG.md:4]: Yep, this is todo. Working on it at the moment.

[the remark this answers is not in the document]
Comment (priitlatt) [reply CHANGELOG.md]: Changelog was added in b2a120aa71d2073edb9acef54ba197ea5e410e77.

**Pull request merged**